### PR TITLE
LB: Modernize and cleanup MetisLB

### PR DESCRIPTION
**Pull request by rbuch**

### CODE REVIEW

File: src/ck-ldb/ckgraph.h [hunk 1]
```diff
@@ -63,10 +63,9 @@ class Edge {
...
       bytes(_bytes) {
     }
-    ~Edge() { }
-    inline int getNeighborId() { return id; }
-    inline int getNumMsgs() { return msgs; }
-    inline int getNumBytes() { return bytes; }
+    inline int getNeighborId() const { return id; }
+    inline int getNumMsgs() const { return msgs; }
+    inline int getNumBytes() const { return bytes; }
     inline void setNumBytes(int _bytes) { bytes = _bytes; }
```
Comment (matthiasdiener) [inline src/ck-ldb/ckgraph.h:69]: No need for `inline` in these functions (same below).

File: src/ck-ldb/MetisLB.C [hunk 2]
```diff
@@ -5,128 +5,126 @@
...
 CreateLBFunc_Def(MetisLB, "Use Metis(tm) to partition object graph")
 
-MetisLB::MetisLB(const CkLBOptions &opt): CBase_MetisLB(opt)
+    MetisLB::MetisLB(const CkLBOptions& opt)
+    : CBase_MetisLB(opt)
 {
   lbname = "MetisLB";
-  if (CkMyPe() == 0 && !quietModeRequested)
-    CkPrintf("CharmLB> MetisLB created.\n");
+  if (CkMyPe() == 0 && !quietModeRequested) CkPrintf("CharmLB> MetisLB created.\n");
```
Comment (matthiasdiener) [inline src/ck-ldb/MetisLB.C:24]: fwiw, I prefer the old way (with a line break before the `CkPrintf`).

File: src/ck-ldb/ckgraph.h (hunk 1 above)
Comment (rbuch) [reply src/ck-ldb/ckgraph.h]: Fixed.

File: src/ck-ldb/MetisLB.C (hunk 2 above)
Comment (rbuch) [reply src/ck-ldb/MetisLB.C]: Fixed.

Comment (rbuch) [reply src/ck-ldb/MetisLB.C]: If this a thing we want to enforce, we should set `AllowShortIfStatementsOnASingleLine` in the Clang-Format configuration correspondingly.

**Review by matthiasdiener — DISMISSED** [review]
LGTM, but maybe clarify in the commit message that this is removing the WEIGHTED option.

**Comment by rbuch** [comment]
> LGTM, but maybe clarify in the commit message that this is removing the WEIGHTED option.

Done.

File: src/ck-ldb/MetisLB.C (hunk 2 above)
Comment (rbuch) [reply src/ck-ldb/MetisLB.C]: Done in #3229.

File: src/ck-ldb/ckgraph.h [hunk 3]
```diff
@@ -14,159 +14,164 @@
...
-    bool available;	// if the processor is available
+public:
+  ProcInfo() = default;
+  ProcInfo(int i, double ov, double tl, double sp, bool avail)
+      : id(i), _overhead(ov), _totalLoad(tl), _pe_speed(sp), available(avail)
+  {
+  }
+  int getProcId() const { return id; }
+  void setProcId(int _id) { id = _id; }
+  double getTotalLoad() const { return _totalLoad; }
```
Comment (epmikida) [inline src/ck-ldb/ckgraph.h:32]: Why two accessors for total load?

**Review by epmikida — DISMISSED** [review]

File: src/ck-ldb/ckgraph.h (hunk 3 above)
Comment (epmikida) [reply src/ck-ldb/ckgraph.h:32]: Also as a followup...do we even want the accessors by reference? are they necessary compared to just get/set methods like are used everywhere else here?

Comment (evan-charmworks) [reply src/ck-ldb/ckgraph.h:32]: Agreed that the accessors need a little bit more cleanup. At the very least, if we want to keep the ones returning by reference, there should be const-qualified versions that return const references. But I'm not sure if we need accessors returning by reference at all.

Comment (rbuch) [reply src/ck-ldb/ckgraph.h:34]: Good call, I've removed the ones that return by reference, I don't think they're needed at all.

**Review by evan-charmworks — APPROVED** [review]

**Review by epmikida — APPROVED** [review]

**Review by stwhite91 — APPROVED** [review]

**Pull request merged**